Examples/test-suite/perl5/li_constraints_runme.pl

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use Test::More tests => 18;
+BEGIN { use_ok('li_constraints') }
+require_ok('li_constraints');
+
+sub check_double {
+    my ($except, $name, $val) = @_;
+    my $fname = "li_constraints::test_$name";
+    my $actual = eval "$fname($val);1";
+    $actual = 0 unless defined $actual;
+    if($actual) {
+      is($actual, $except, "$fname pass with $val");
+    } else {
+      is($actual, $except, "$fname throw exception with $val");
+    }
+}
+check_double(1, "nonnegative", 10);
+check_double(1, "nonnegative", 0);
+check_double(0, "nonnegative", -10);
+
+check_double(0, "nonpositive", 10);
+check_double(1, "nonpositive", 0);
+check_double(1, "nonpositive", -10);
+
+check_double(1, "positive", 10);
+check_double(0, "positive", 0);
+check_double(0, "positive", -10);
+
+check_double(0, "negative", 10);
+check_double(0, "negative", 0);
+check_double(1, "negative", -10);
+
+check_double(1, "nonzero", 10);
+check_double(0, "nonzero", 0);
+check_double(1, "nonzero", -10);
+
+# Pass null value
+my $ret = eval "li_constraints::test_nonnull(undef); 1";
```

Comment:
`eval` with a block would be better Perl I think as then the code is only parsed once before the script starts., i.e.:

```
my $ret = eval { li_constraints::test_nonnull(undef); 1 };
```

Similarly above.

Reply:
Oh, except the above one generates the code so that can't just be a block.

Reply:
It works on my machine.
What do you prefer?

Reply:
It works, but I think the block form is better when you can use it (as in this case).

Reply:
I replace the string evaluation with function referee or lambada functions in all scripts.